src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/queryAllPackages.hpp

```diff
@@ -0,0 +1,254 @@
+                        package.value("format", "").c_str(),
+                        package.value("groups", "").c_str(),
+                        package.value("install_time", "").c_str(),
+                        package.value("item_id", "").c_str(),
+                        package.value("location", "").c_str(),
+                        package.value("multiarch", "").c_str(),
+                        package.value("name", "").c_str(),
+                        package.value("priority", "").c_str(),
+                        package.value("scan_time", "").c_str(),
+                        package.value("size", 0),
+                        package.value("source", "").c_str(),
+                        package.value("vendor", "").c_str(),
+                        package.value("version", "").c_str());
+
+                    auto stateMsg = SyscollectorSynchronization::Createstate(
+                        fbBuilder,
+                        SyscollectorSynchronization::AttributesUnion::AttributesUnion_syscollector_packages,
+                        syscollectorPackage.Union());
+
+                    auto syncMsg = SyscollectorSynchronization::CreateSyncMsg(
+                        fbBuilder,
+                        agentInfo,
+                        SyscollectorSynchronization::DataUnion::DataUnion_state,
+                        stateMsg.Union());
+
+                    fbBuilder.Finish(syncMsg);
+
+                    std::variant<const SyscollectorDeltas::Delta*,
+                                 const SyscollectorSynchronization::SyncMsg*,
+                                 const nlohmann::json*>
+                        variantData = SyscollectorSynchronization::GetSyncMsg(fbBuilder.GetBufferPointer());
+
+                    // This could throw a exception if there is not information about the os, because the user disable
+                    // it from syscollector.
+                    try
+                    {
+                        auto context = std::make_shared<TScanContext>(variantData);
+                        m_suborchestration->handleRequest(std::move(context));
+                    }
+                    catch (const std::exception& e)
+                    {
+                        logError(WM_VULNSCAN_LOGTAG, "Error handling request: %s", e.what());
+                    }
+                }
+            }
+            else
+            {
+                logWarn(WM_VULNSCAN_LOGTAG,
+                        "Empty response for agent '%s' in Wazuh-DB 'sys_programs' query",
+                        std::string(data->agentId()).c_str());
+            }
+        }
+        else if (data->operationType() == OperationType::InsertAll)
+        {
+            logDebug2(WM_VULNSCAN_LOGTAG, "Re-scanning packages for all agents");
+            for (const auto& agent : data->getMessages())
+            {
+                // Execute query for all packages, modify to adapt to testtool input.
```

Comment:
Another option was simply use the Id as an integer in the test tool data files, because that't the way it's in the `global.db`

Reply:
Is not possible because an integer can not be key in the JSON object.

Reply:
Ok, thank you.